src/Std/Internal/Async/UDP.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time
+import Std.Internal.UV
+import Std.Internal.Async.Basic
+import Std.Net.Addr
+
+namespace Std
+namespace Internal
+namespace IO
+namespace Async
+namespace UDP
+
+open Std.Net
+
+/--
+Represents a UDP socket.
+-/
+structure Socket where
+  private ofNative ::
+    native : Internal.UV.UDP.Socket
+
+/--
+Membership type for multicast operations.
+-/
+inductive Membership
+  | leaveGroup
+  | enterGroup
+
+namespace Socket
+
+/--
+Creates a new UDP socket.
+-/
+@[inline]
+def mk : IO Socket := do
+  let native ← Internal.UV.UDP.Socket.new
+  return Socket.ofNative native
+
+/--
+Binds the UDP socket to the given address.
+-/
+@[inline]
+def bind (s : Socket) (addr : SocketAddress) : IO Unit :=
+  s.native.bind addr
+
+/--
+Connects the UDP socket to the given address.
+-/
+@[inline]
+def connect (s : Socket) (addr : SocketAddress) : IO Unit :=
+  s.native.connect addr
+
+/--
+Sends data through the UDP socket.
+-/
+@[inline]
+def send (s : Socket) (data : ByteArray) (addr : Option SocketAddress := none) : IO (AsyncTask Unit) :=
+  AsyncTask.ofPromise <$> s.native.send data addr
+
+/--
+Receives data from the UDP socket.
+-/
+@[inline]
+def recv (s : Socket) (size : UInt64) : IO (AsyncTask (ByteArray × SocketAddress)) :=
+  AsyncTask.ofPromise <$> s.native.recv size
+
+/--
+Gets the local address of the UDP socket.
+-/
+@[inline]
+def getSockName (s : Socket) : IO SocketAddress :=
+  s.native.getSockName
+
+/--
+Gets the remote address of the UDP socket.
+-/
```

Comment:
Should mention the connection to `connect` here and what happens if `connect` wasn't called.